cns/restserver/util.go

```diff
@@ -756,12 +756,12 @@ func (service *HTTPRestService) SendNCSnapShotPeriodically(ctx context.Context,
 }
 
 func (service *HTTPRestService) validateIPConfigsRequest(ctx context.Context, ipConfigsRequest cns.IPConfigsRequest) (cns.PodInfo, types.ResponseCode, string) {
-	if service.state.OrchestratorType != cns.KubernetesCRD && service.state.OrchestratorType != cns.Kubernetes {
-		return nil, types.UnsupportedOrchestratorType, "ReleaseIPConfig API supported only for kubernetes orchestrator"
+	if service.state.OrchestratorType != cns.KubernetesCRD && service.state.OrchestratorType != cns.Kubernetes && service.state.OrchestratorType != cns.ServiceFabric {
```

Comment:
do we really need this validation as it supported for both k8s and standalone case

Reply:
Yes, we have to have this validation in SF

Reply:
better to have all supported cases here

Reply:
i still don't get the reason why its required to have this validation check. what's the impact if we remove this validation?

Reply:
if we remove this check, then when we test in the SF environment, it will generate this issue ""ReleaseIPConfig API supported only for kubernetes orchestrator""

Reply:
no what happens if we remove condition all together as it supported both for k8s and standalone? whats the significance of this check? cc: @nddq

Reply:
this predates my time here, but I'm guessing we used to not support other orchestrator type within this data path, and this is used to prevent them to invoking this API. But with the introduction of the middleware, other orchestrators can define their own data path to get and release IP configs, so you are right that we might be able get away with removing this all together? correct me if im wrong @rbtr

Reply:
> what's the impact if we remove this validation?

IMO, nothing, and we can remove it entirely. 
If you're running CNS in a scenario where you don't need to call Request/ReleaseIP, and you call it...that's mostly on you. I don't need extra validation in this handler to make sure you have started CNS in the right Mode. You're going to get _an error_ whether we check Mode here or not. And if we are gonna keep growing the list of acceptable modes in this check, we should just let it pass by default and let CNS fail the request because something else like the payload will be wrong.
The better way to accomplish this sort of Mode/API surface gating would be to only enable the Request/ReleaseIP routes when we are running in a supported Mode. Then the rest of the time, those APIs would just not be wired up and would 404. But our custom server mux is not that adjustable...